src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/settings/LanguageServerConfigurable.java

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.intellij.quarkus.lsp4ij.settings;
+
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.ui.NamedConfigurable;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.openapi.util.NlsContexts;
+import com.redhat.devtools.intellij.quarkus.lsp4ij.LanguageServersRegistry;
+
+import javax.swing.*;
+
+/**
+ * UI settings to configure a given language server:
+ *
+ * <ul>
+ *     <li>Debug port</li>
+ *     <li>Suspend and wait for a debugger</li>
+ * </ul>
+ */
+public class LanguageServerConfigurable extends NamedConfigurable<LanguageServersRegistry.LanguageServerDefinition> {
+
+    private final LanguageServersRegistry.LanguageServerDefinition languageServerDefinition;
+
+    private LanguageServerView myView;
+
+    public LanguageServerConfigurable(LanguageServersRegistry.LanguageServerDefinition languageServerDefinition, Runnable updater) {
+        super(false, updater);
+        this.languageServerDefinition = languageServerDefinition;
+    }
+
+    @Override
+    public void setDisplayName(String name) {
+
+    }
+
+    @Override
+    public LanguageServersRegistry.LanguageServerDefinition getEditableObject() {
+        return languageServerDefinition;
+    }
+
+    @Override
+    public @NlsContexts.DetailedDescription String getBannerSlogan() {
+        return languageServerDefinition.getDisplayName();
+    }
+
+    @Override
+    public JComponent createOptionsPanel() {
+        if (myView == null) {
+            myView = new LanguageServerView(languageServerDefinition);
+        }
+        return myView.getComponent();
+    }
+
+    @Override
+    public @NlsContexts.ConfigurableName String getDisplayName() {
+        return languageServerDefinition.getDisplayName();
+    }
+
+    @Override
+    public boolean isModified() {
+        LanguageServerSettingsState.LanguageServerDefinitionSettings settings = LanguageServerSettingsState.getInstance()
+                .getLanguageServerSettings(languageServerDefinition.id);
+        if (settings == null) {
+            return true;
+        }
+        boolean modified = !myView.getDebugPort().equals(settings.getDebugPort());
+        modified = modified | myView.isDebugSuspend() != settings.isDebugSuspend();

```

Comment:
You don't need/want the right-hand side to be evaluated in this case. There is *practically* never a good reason to use single `|` over short-circuit `||`. 
See https://stackoverflow.com/a/7102013/753170

Reply:
My bad, I wanted to mimic https://github.com/redhat-developer/intellij-tekton/blob/623343da0bd715cdfa0ab023066975e99b588cd4/src/main/java/com/redhat/devtools/intellij/tektoncd/settings/SettingsConfigurable.java#L38

Now I'm using ||